### PR TITLE
Separate rerun build naming by rerun of and rerun number

### LINKED ISSUE #4622: API: more structured API for re-ran builds
* Don't bake dot-notation into the name
* Extract "run number" (i.e. the 2 in 241.2) into a separate column
* Return the build that it's a re-run of (consult UX to see what's useful here)

### PR DESCRIPTION
**Pull request by clarafu**
We return back the rerun of name and rerun number in the API response for a build in order to allow the web UI more flexibility in formatting the naming.

Also sneaks in bumping of the migrations.. :)

fixes #4622 

# Contributor Checklist
> Are the following items included as part of this PR? Please delete checkbox items that don't apply.
- [ ] Unit tests
- [ ] Integration tests (if applicable)
- [ ] Updated documentation (located at https://github.com/concourse/docs)
- [ ] Updated release notes (located at https://github.com/concourse/concourse/tree/master/release-notes)

# Reviewer Checklist
> This section is intended for the core maintainers only, to track review progress. Please do not
> fill out this section.
- [ ] Code reviewed
- [ ] Tests reviewed
- [ ] Documentation reviewed
- [ ] Release notes reviewed
- [ ] PR acceptance performed
- [ ] New config flags added? Ensure that they are added to the [BOSH](https://github.com/concourse/concourse-bosh-release) 
      and [Helm](https://github.com/concourse/helm) packaging; otherwise, ignored for the [integration tests](https://github.com/concourse/ci/tree/master/tasks/scripts/check-distribution-env) (for example, if they are Garden configs that are not displayed in the `--help` text).

### CODE REVIEW

File: atc/api/jobs_test.go [hunk 1]
```diff
@@ -1404,6 +1417,19 @@ var _ = Describe("Jobs API", func() {
 						"team_name": "some-team",
 						"start_time": 101,
 						"end_time": 200
+					},
+					{
+						"id": 3,
+						"name": "1.1",
```
Comment (vito) [inline atc/api/jobs_test.go]: Might want to make this `1.2` for clarity.

File: atc/api/jobs_test.go [hunk 2]
```diff
@@ -1404,6 +1417,19 @@ var _ = Describe("Jobs API", func() {
...
+						"job_name": "some-job",
+						"status": "succeeded",
+						"api_url": "/api/v1/builds/3",
+						"pipeline_name": "some-pipeline",
+						"team_name": "some-team",
+						"start_time": 102,
+						"end_time": 300,
+						"rerun_of": "1",
```
Comment (vito) [inline atc/api/jobs_test.go]: I don't think it's super obvious what this value is. What about something like this?:

```json
{
  "rerun_of": {
    "id": 1234,
    "name": "1",
    "api_url": "/api/v1/builds/2"
  }
}
```

File: atc/api/jobs_test.go [hunk 3]
```diff
@@ -1404,6 +1417,19 @@ var _ = Describe("Jobs API", func() {
...
+						"rerun_number": "2"
```
Comment (vito) [inline atc/api/jobs_test.go]: I would expect this to be an integer, not a string.

File: atc/db/build.go [hunk 4]
```diff
@@ -1246,6 +1254,17 @@ func (b *build) AdoptRerunInputsAndPipes() ([]BuildInput, bool, error) {
 		return nil, false, err
 	}
 
+	_, err = psql.Update("builds").
```
Comment (vito) [inline atc/db/build.go]: This seems out-of-place in this PR; is it fixing something? Should it be a separate commit + test?

File: atc/api/jobs_test.go (hunk 1 above)
Comment (clarafu) [reply atc/api/jobs_test.go]: Wait why would that be 1.2?

Comment (vito) [reply atc/api/jobs_test.go]: Because the rerun number is 2, not 1

Comment (clarafu) [reply atc/api/jobs_test.go]: Oh the rerun number should be 1 then

File: atc/build.go [hunk 5]
```diff
@@ -12,16 +12,23 @@ const (
...
+	TeamName     string        `json:"team_name"`
+	Name         string        `json:"name"`
+	Status       string        `json:"status"`
+	JobName      string        `json:"job_name,omitempty"`
+	APIURL       string        `json:"api_url"`
+	PipelineName string        `json:"pipeline_name,omitempty"`
+	StartTime    int64         `json:"start_time,omitempty"`
+	EndTime      int64         `json:"end_time,omitempty"`
+	ReapTime     int64         `json:"reap_time,omitempty"`
+	RerunNumber  string        `json:"rerun_number,omitempty"`
```
Comment (vito) [inline atc/build.go]: I think this should be an `int`.

File: atc/db/job.go [hunk 6]
```diff
@@ -924,20 +925,25 @@ func (j *job) finishedBuild(tx Tx) (Build, error) {
...
 		RunWith(tx).
 		QueryRow().
-		Scan(&buildName, &rerunName)
+		Scan(&buildName, &rerunNum)
+	if err != nil {
+		return "", 0, err
+	}
+
+	// increment the rerun number
+	rerunNum++
```
Comment (pivotal-bin-ju) [inline atc/db/job.go]: just wondering whether we need to update the `rerun_number` here?
